Song.js

```diff
@@ -29,6 +30,24 @@ class Song
 			this.makePeriscopeStream(callback);
 		} else if (url.endsWith(".mp3") && (url.startsWirth("http://") || url.startsWith("https://"))) {
 			this.makeMP3Stream(callback);
+		} else if (config.youtube && config.youtube.token) {
+			console.log("Searching in YouTube: " + url);
+			var options = {
+				maxResults: 1,
+				key: config.youtube.token
+			};
+			yt_search(url, options, (error, results) => {
```

Comment:
`url` here should probably be renamed to `query` if you changed it in the caller.

Reply:
This would require changing `url` to `query` in `Song` constructor, plus probably everywhere else (`SongQueue` methods and so on).